Build fails with nginx 1.23.0

```
cc -c -fPIC -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -fPIC -D_GNU_SOURCE -I src/core -I src/event -I src/e
 -o objs/addon/ngx_http_geoip2_module/ngx_http_geoip2_module.o \
 ../ngx_http_geoip2_module/ngx_http_geoip2_module.c
../ngx_http_geoip2_module/ngx_http_geoip2_module.c: In function 'ngx_http_geoip2_variable':
../ngx_http_geoip2_module/ngx_http_geoip2_module.c:172:14: error: assignment to 'ngx_array_t *' from incompatible pointer type 'ngx_table_elt_t **' {aka 'struct ngx_table_elt_s **'} [-Werror=incompatible-pointer-types]
  172 |         xfwd = &r->headers_in.x_forwarded_for;
      |              ^
../ngx_http_geoip2_module/ngx_http_geoip2_module.c:175:58: error: passing argument 3 of 'ngx_http_get_forwarded_addr' from incompatible pointer type [-Werror=incompatible-pointer-types]
  175 |             (void) ngx_http_get_forwarded_addr(r, &addr, xfwd, NULL,
      |                                                          ^~~~
      |                                                          |
      |                                                          ngx_array_t *
In file included from src/http/ngx_http.h:36,
                 from ../ngx_http_geoip2_module/ngx_http_geoip2_module.c:10:
src/http/ngx_http_core_module.h:532:22: note: expected 'ngx_table_elt_t *' {aka 'struct ngx_table_elt_s *'} but argument is of type 'ngx_array_t *'
  532 |     ngx_table_elt_t *headers, ngx_str_t *value, ngx_array_t *proxies,
      |     ~~~~~~~~~~~~~~~~~^~~~~~~
cc1: all warnings being treated as errors
```

--- a/ngx_http_geoip2_module.c
+++ b/ngx_http_geoip2_module.c
@@ -146,7 +146,11 @@ ngx_http_geoip2_variable(ngx_http_request_t *r, ngx_http_variable_value_t *v,
     MMDB_entry_data_s       entry_data;
     ngx_http_geoip2_conf_t  *gcf;
     ngx_addr_t              addr;
+#if defined(nginx_version) && nginx_version >= 1023000
+    ngx_table_elt_t         *xfwd;
+#else
     ngx_array_t             *xfwd;
+#endif
     u_char                  *p;
     ngx_str_t               val;
 
@@ -169,9 +173,15 @@ ngx_http_geoip2_variable(ngx_http_request_t *r, ngx_http_variable_value_t *v,
         addr.sockaddr = r->connection->sockaddr;
         addr.socklen = r->connection->socklen;
 
+#if defined(nginx_version) && nginx_version >= 1023000
+        xfwd = r->headers_in.x_forwarded_for;
+
+        if (xfwd != NULL && gcf->proxies != NULL) {
+#else
         xfwd = &r->headers_in.x_forwarded_for;
 
         if (xfwd->nelts > 0 && gcf->proxies != NULL) {
+#endif
             (void) ngx_http_get_forwarded_addr(r, &addr, xfwd, NULL,
                                                gcf->proxies, gcf->proxy_recursive);
         }

